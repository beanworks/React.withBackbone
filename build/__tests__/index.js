'use strict';var _extends=Object.assign||function(a){for(var c,b=1;b<arguments.length;b++)for(var d in c=arguments[b],c)Object.prototype.hasOwnProperty.call(c,d)&&(a[d]=c[d]);return a},_createClass=function(){function a(b,c){for(var e,d=0;d<c.length;d++)e=c[d],e.enumerable=e.enumerable||!1,e.configurable=!0,'value'in e&&(e.writable=!0),Object.defineProperty(b,e.key,e)}return function(b,c,d){return c&&a(b.prototype,c),d&&a(b,d),b}}(),_index=require('../index.js'),_propTypes=require('prop-types'),_propTypes2=_interopRequireDefault(_propTypes),_react=require('react'),_react2=_interopRequireDefault(_react),_reactTestRenderer=require('react-test-renderer'),_reactTestRenderer2=_interopRequireDefault(_reactTestRenderer),_testUtils=require('react-dom/test-utils'),_testUtils2=_interopRequireDefault(_testUtils),_backbone=require('backbone'),_backbone2=_interopRequireDefault(_backbone);function _interopRequireDefault(a){return a&&a.__esModule?a:{default:a}}function _classCallCheck(a,b){if(!(a instanceof b))throw new TypeError('Cannot call a class as a function')}function _possibleConstructorReturn(a,b){if(!a)throw new ReferenceError('this hasn\'t been initialised - super() hasn\'t been called');return b&&('object'==typeof b||'function'==typeof b)?b:a}function _inherits(a,b){if('function'!=typeof b&&null!==b)throw new TypeError('Super expression must either be null or a function, not '+typeof b);a.prototype=Object.create(b&&b.prototype,{constructor:{value:a,enumerable:!1,writable:!0,configurable:!0}}),b&&(Object.setPrototypeOf?Object.setPrototypeOf(a,b):a.__proto__=b)}describe('When using withBackbone',function(){var a=function(k){function l(m){return _classCallCheck(this,l),_possibleConstructorReturn(this,(l.__proto__||Object.getPrototypeOf(l)).call(this,m))}return _inherits(l,k),_createClass(l,[{key:'instanceMethod',value:function instanceMethod(){return'Instance Method was Invoked!'}},{key:'render',value:function render(){return _react2.default.createElement('div',{id:'testDiv'},_react2.default.createElement('p',{id:'A'},this.props.user.get('name'),' - ',this.props.job.get('name')),this.props.cars.models.map(function(m){return _react2.default.createElement('p',{key:m.cid},' ',m.get('brand'),' - ',m.get('model'))}),this.props.cities.models.map(function(m){return _react2.default.createElement('p',{key:m.cid},' ',m.get('name'),' - ',m.get('weather'))}))}}]),l}(_react2.default.Component);a.propTypes={cars:_propTypes2.default.instanceOf(_backbone2.default.Collection),cities:_propTypes2.default.instanceOf(_backbone2.default.Collection),user:_propTypes2.default.instanceOf(_backbone2.default.Model),job:_propTypes2.default.instanceOf(_backbone2.default.Model)};var b=(0,_index.withBackbone)(a),c=function(k){function l(m){_classCallCheck(this,l);var n=_possibleConstructorReturn(this,(l.__proto__||Object.getPrototypeOf(l)).call(this,m));return n.state={user:new _backbone2.default.Model,job:new _backbone2.default.Model,cars:new _backbone2.default.Collection,cities:new _backbone2.default.Collection},n}return _inherits(l,k),_createClass(l,[{key:'render',value:function render(){var m=this;return _react2.default.createElement(b,_extends({},this.state,{ref:function ref(n){m.viewUnderTest=n}}))}}]),l}(_react2.default.Component),d,e,f,g,h,j;beforeEach(function(){e=new _backbone2.default.Model({name:'Tom'}),f=new _backbone2.default.Model({name:'developer'}),g=new _backbone2.default.Collection([{brand:'Ford',model:'F150'},{brand:'Jeep',model:'Wrangler'},{brand:'Tesla',model:'X'}]),h=new _backbone2.default.Collection([{name:'Vancouver',weather:'rainy'},{name:'Victoria',weather:'windy'},{name:'Whistler',weather:'SNOW!!!'}]),d=_react2.default.createElement(c),j=_testUtils2.default.renderIntoDocument(d),j.setState({user:e,job:f,cars:g,cities:h})}),it('Should render match to snapshot',function(){var k=j.render(),l=_reactTestRenderer2.default.create(k);expect(l).toMatchSnapshot()}),it('Should be able to handle more than one model at a time and trigger rerender when the model changed',function(){var k=_testUtils2.default.scryRenderedDOMComponentsWithTag(j,'p')[0];expect(k.textContent).toEqual('Tom - developer'),e.set('name','Apple'),j.setState({user:e}),k=_testUtils2.default.scryRenderedDOMComponentsWithTag(j,'p')[0],expect(k.textContent).toEqual('Apple - developer'),f.set('name','Engineer'),j.setState({job:f}),k=_testUtils2.default.scryRenderedDOMComponentsWithTag(j,'p')[0],expect(k.textContent).toEqual('Apple - Engineer')}),it('Should be able to handle more than one collection at a time and trigger rerender when the collection resets',function(){g.reset();var k=j.render(),l=_reactTestRenderer2.default.create(k);expect(l).toMatchSnapshot(),h.add({name:'Montreal',weather:'sunny'}),k=j.render(),l=_reactTestRenderer2.default.create(k),expect(l).toMatchSnapshot()}),it('Should be able to access wrappedComponent\'s instance method via ref',function(){expect(j.viewUnderTest).not.toBeFalsy(),expect(j.viewUnderTest.wrappedComponent).not.toBeFalsy(),expect(j.viewUnderTest.wrappedComponent.instanceMethod()).toEqual('Instance Method was Invoked!')}),it('Should register listener for models and collections',function(){expect(Object.keys(j.viewUnderTest.collectionListener._listeningTo).length).toBe(2),expect(Object.keys(j.viewUnderTest.modelListener._listeningTo).length).toBe(2)})});