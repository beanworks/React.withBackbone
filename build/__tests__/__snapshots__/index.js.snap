// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using withBackbone Should be able to handle more than one collection at a time and trigger rerender when the collection resets 1`] = `
<div
  id="testDiv"
>
  <p
    id="A"
  >
    Tom
     - 
    developer
  </p>
  <p>
     
    Vancouver
     - 
    rainy
  </p>
  <p>
     
    Victoria
     - 
    windy
  </p>
  <p>
     
    Whistler
     - 
    SNOW!!!
  </p>
</div>
`;

exports[`When using withBackbone Should be able to handle more than one collection at a time and trigger rerender when the collection resets 2`] = `
<div
  id="testDiv"
>
  <p
    id="A"
  >
    Tom
     - 
    developer
  </p>
  <p>
     
    Vancouver
     - 
    rainy
  </p>
  <p>
     
    Victoria
     - 
    windy
  </p>
  <p>
     
    Whistler
     - 
    SNOW!!!
  </p>
  <p>
     
    Montreal
     - 
    sunny
  </p>
</div>
`;

exports[`When using withBackbone Should render match to snapshot 1`] = `
<div
  id="testDiv"
>
  <p
    id="A"
  >
    Tom
     - 
    developer
  </p>
  <p>
     
    Ford
     - 
    F150
  </p>
  <p>
     
    Jeep
     - 
    Wrangler
  </p>
  <p>
     
    Tesla
     - 
    X
  </p>
  <p>
     
    Vancouver
     - 
    rainy
  </p>
  <p>
     
    Victoria
     - 
    windy
  </p>
  <p>
     
    Whistler
     - 
    SNOW!!!
  </p>
</div>
`;
